src/components/balances/HiddenAssetsProvider/index.tsx

```diff
@@ -0,0 +1,119 @@
+import useBalances from '@/hooks/useBalances'
+import useChainId from '@/hooks/useChainId'
+import useHiddenAssets from '@/hooks/useHiddenAssets'
+import { useAppDispatch } from '@/store'
+import { addHiddenAssets, removeHiddenAssets } from '@/store/hiddenAssetsSlice'
+import { type SafeBalanceResponse, TokenType } from '@safe-global/safe-gateway-typescript-sdk'
+import { type ReactElement, type ReactNode, useState, useCallback, useMemo, createContext } from 'react'
+
+type HiddenAssetsContextValue = {
+  showHiddenAssets: boolean
+  toggleShowHiddenAssets: () => void
+  toggleAsset: (address: string) => void
+  saveChanges: () => void
+  reset: () => void
+  isAssetSelected: (address: string) => boolean
+  visibleAssets: SafeBalanceResponse['items']
+}
+
+export const HiddenAssetsContext = createContext<HiddenAssetsContextValue>({
+  showHiddenAssets: false,
+  toggleShowHiddenAssets: () => {},
+  toggleAsset: () => {},
+  saveChanges: () => {},
+  reset: () => {},
+  isAssetSelected: () => true,
+  visibleAssets: [],
+})
+
+const HiddenAssetsProvider = ({ children }: { children: ReactNode }): ReactElement => {
```

Comment:
Why do you need a provider when we usually just use selectors and dispatchers directly? Feels like an unnecessary abstraction layer.